Unable to compile cpp file generated from _pixelfunc.pyx

While trying to build from source I found the following error:

```
877      healpy/src/_pixelfunc.cpp: In function 'PyObject* __pyx_pf_6healp
            y_10_pixelfunc_ringinfo(PyObject*, PyObject*, PyArrayObject*)':
  >> 878    healpy/src/_pixelfunc.cpp:2185:32: error: 'PyArrayObject' {aka 'str
            uct tagPyArrayObject'} has no member named 'dimensions'
     879       2185 |   __pyx_v_num = (__pyx_v_ring->dimensions[0]);
     880            |                                ^~~~~~~~~~
```

It appears cython generated code which uses a numpy API which is no longer supported.  Removing the NPY_NO_DEPRECATED_API option (https://github.com/healpy/healpy/blob/main/setup.py#L316) caused a few warnings but successfully built.  Likewise setting the value to `None` works as well.

Here is my environment:

```
$ python -V
Python 3.9.7
$ python -c 'import numpy; print(numpy.__version__)'
1.19.5
$ cython --version
Cython version 0.29.36
$ git log -n 1
commit e952ac4fddcaa187d23f97a12859f4c222aea164 (HEAD -> main, origin/main, origin/HEAD)
Author: Leo Singer <leo.singer@ligo.org>
Date:   Thu Mar 14 11:39:47 2024 -0400

    Move vendored sources to cextern directory

    This source layout is more similar to Astropy affiliated packages.
ubuntu@ip-172-31-47-127:/fsx/lrbison/code/healpy$
```

And my build command

```
python setup.py build_ext
```

--- a/INSTALL.rst
+++ b/INSTALL.rst
@@ -216,19 +216,14 @@ Developers building from a snapshot of the github repository need:
 * ``libssl-dev`` (Debian) or ``openssl-dev`` (CentOS)
   is required to build ``cfitsio`` from source
 
-* `cython` > 0.16
-
 * run ``git submodule init`` and ``git submodule update`` to get the bundled
   HEALPix sources
 
-the best way to install healpy if you plan to develop is to build the C++
-extensions in place with::
+The best way to install healpy if you plan to develop is to do a
+`development mode (a.k.a. "editable" install) <https://setuptools.pypa.io/en/latest/userguide/development_mode.html>`_
+with pip by adding the ``-e`` flag::
 
-    python setup.py build_ext --inplace
-
-then add the ``healpy`` repository folder to your ``PYTHONPATH`` (e.g. if you
-cloned this repository to ``$REPOS`` such that ``$REPOS/healpy/INSTALL.rst``
-exists, then add ``$REPOS/healpy`` to your ``PYTHONPATH``).
+    pip install -e .
 
 In case of compilation errors, see the note above in the ``pip`` section.
 

